pkg/resource/hyper_parameter_tuning_job/custom_set_output.go

```diff
@@ -0,0 +1,106 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Use this file if the Status/Spec of the CR needs to be modified after
+// create/describe/update operation
+
+package hyper_parameter_tuning_job
+
+import (
+	"context"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// customCreateHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in creating state. At this stage we know call to createHyperParameterTuningJob was successful.
+func (rm *resourceManager) customCreateHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.CreateHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.HyperParameterTuningJobStatusInProgress), ko)
+	return ko, nil
+}
+
+// customDescribeHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if
+// HyperParameterTuningJob is being modified by AWS.
+func (rm *resourceManager) customDescribeHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.DescribeHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, resp.HyperParameterTuningJobStatus, ko)
+	return ko, nil
+}
+
+// customStopHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in stopping state.
+func (rm *resourceManager) customStopHyperParameterTuningJobSetOutput(
```

Comment:
unreachable code, doesnt look like sdkDelete has a customSetOutput. see sdk.go

Also, I don't think we need this because the resource is marked as unmanaged as soon as delete is successful

https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/runtime/reconciler.go#L317
https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/runtime/reconciler.go#L325

Reply:
Agreed, I had put this in description but was not absolutely sure if it was to remove. Deleting.